From 8f59c9b4858cde1352eb60b2a3fe22853813b287 Mon Sep 17 00:00:00 2001
From: Kohei Yoshida <kohei.yoshida@collabora.com>
Date: Mon, 30 Jun 2014 14:01:16 -0400
Subject: [PATCH 1/3] Check for empty() before calling top().

Else it might crash sometimes.

Change-Id: I6a24fff83c3d36346debae5c0f2b8c0646a15c01
(cherry picked from commit 3f177756dbdb67d901453000c3f11694770d2761)
Reviewed-on: https://gerrit.libreoffice.org/9991
Tested-by: Michael Stahl <mstahl@redhat.com>
Reviewed-by: Michael Stahl <mstahl@redhat.com>
---
 sax/source/fastparser/fastparser.cxx | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/sax/source/fastparser/fastparser.cxx b/sax/source/fastparser/fastparser.cxx
index 0cfdb14..577f414 100644
--- a/sax/source/fastparser/fastparser.cxx
+++ b/sax/source/fastparser/fastparser.cxx
@@ -682,6 +682,9 @@ sal_Int32 FastSaxParserImpl::GetTokenWithPrefix( const sal_Char*pPrefix, int nPr
     sal_Int32 nNamespaceToken = FastToken::DONTKNOW;
 
     Entity& rEntity = getEntity();
+    if (rEntity.maNamespaceCount.empty())
+        return nNamespaceToken;
+
     sal_uInt32 nNamespace = rEntity.maNamespaceCount.top();
     while( nNamespace-- )
     {
-- 
1.9.3

